src/CrossScopeValidator.ts

```diff
@@ -461,9 +461,10 @@ export class CrossScopeValidator {
     }
 
     clearResolutionsForScopes(scopes: Scope[]) {
+        const lowerScopeNames = scopes.map(scope => scope.name.toLowerCase());
```

Comment:
Switching this to a `Set` because that's significantly faster when looking something up.

https://jsperf.app/pikuzu/7

Reply:
@markwpearce just thinking about general performance. We're going to run `.toLowerCase()` on every scope name every time we do this logic (and probably other logic elsewhere). It would probably be a good idea to store a permanent lower-case name for the scope (or just always force scope.name to _be_ lower case permanently). Not something to solve here, but just a thought. I'll open a ticket to track this so we can talk about it.